### ISSUE TITLE
Question about signature_cache parameter

### ISSUE THREAD

**Issue by yihming** [MaayanLab/blitzgsea#31]
Hello,

When `signature_cache=True`, which is by default, it seems that the caching uses the gene signature vector as the hash key. My question is, if I then try a different library but on the same signature, should I set `signature_cache=False` in order to get the more accurate and reproducible results? E.g.

```python
import blitzgsea as blitz
res1 = blitz.gsea(signature, library1, seed=0)
res2 = blitz.gsea(signature, library2, seed=0, signature_cache=False)
```

Thanks!

Sincerely,
Yiming

**Comment by lachmann12** [comment]
Hi,

the signature_cache is reused if the distribution of values in a signature are identical. The reason is that the null model to calculate the normalized enrichment scores depends on the size of the gene-set and the value distribution from the signature. If you switch the library it does not matter, since blitzgsea calculates null model parameters for a large range of gene set sizes. So it can reuse previously calculated null models for the same signature, even if the library changes.

**Comment by yihming** [comment]
Thanks for the explanation.

The reason I'm asking this question is that I encounter the case that my results would be different if the same signature runs with `lib1` then `lib2`, comparing to only running with `lib2`.

Namely, the following code
```
_ = blitz.gsea(signature, lib1, seed=0)
res = blitz.gsea(signature, lib2, seed=0)
```
would give different `res` than
```
res = blitz.gsea(signature, lib2, seed=0)
```
while by setting `signature_cache=False` solves the issue:
```
_ = blitz.gsea(signature, lib1, seed=0)
res = blitz.gsea(signature, lib2, seed=0, signature_cache=False)
```

Since `library` is sort of used when estimating the anchors (https://github.com/MaayanLab/blitzgsea/blob/main/blitzgsea/__init__.py#L111-L112), I wonder if that is the reason for the different results.

**Comment by lachmann12** [comment]
There is definitely some randomization that can cause slightly different results to random permutation sampling. How different are the results?

**Comment by yihming** [comment]
The `es`, `geneset_size` and `leading_edge` columns remain the same, while `nes`, `pval`, `sidak` and `fdr` are different.

I checked the 670 terms in the results, and 314 of them has adjusted p-values (`fdr`) differ larger than 0.1. This may change their significance at FDR 10% level. I've listed some of those terms below.

<img width="1186" alt="Screenshot 2024-06-04 at 15 00 23" src="https://github.com/MaayanLab/blitzgsea/assets/2455754/cdc6d937-4187-4ac6-85fc-14dc1f327598">
